.github/workflows/main.yml

```diff
@@ -12,32 +12,34 @@ jobs:
       fail-fast: false
       matrix:
         os: [ubuntu-latest]
-        python-version: ["3.8", "3.9"]
+        python-version: ["3.9", "3.10"]
         numpy_ver: [latest]
         include:
-          - python-version: "3.7"
-            numpy_ver: "1.18"
+          - python-version: "3.8"
+            numpy_ver: "1.20"
             os: ubuntu-latest
 
     name: Python ${{ matrix.python-version }} on ${{ matrix.os }} with numpy ${{ matrix.numpy_ver }}
     runs-on: ${{ matrix.os }}
     steps:
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@v3
     - name: Set up Python ${{ matrix.python-version }}
-      uses: actions/setup-python@v2
+      uses: actions/setup-python@v4
       with:
         python-version: ${{ matrix.python-version }}
 
-    - name: Install standard dependencies
+    - name: Install with standard dependencies
+      if: ${{ matrix.numpy_ver == 'latest'}}
       run: |
-        python -m pip install --upgrade pip
-        pip install -r requirements.txt
-        pip install -r test_requirements.txt
+        pip install .[test]
 
-    - name: Install NEP29 dependencies
+    - name: Install with NEP29 dependencies
       if: ${{ matrix.numpy_ver != 'latest'}}
       run: |
         pip install --no-binary :numpy: numpy==${{ matrix.numpy_ver }}
+        # Pandas 1.5 does not support NEP29
+        pip install "pandas<1.5"
+        pip install  --upgrade-strategy only-if-needed .[test]
```

Comment:
Need to force install any packages that do not support the minimum numpy.  The upgrade strategy will keep pip from updating numpy.